KonyAccelerate/controllers/mobile/frmEventLocationController/frmEventLocationController.js

```diff
@@ -1,30 +1,84 @@
+            "$filter": "((SoftDeleteFlag ne true) or (SoftDeleteFlag eq null))",
+            "$expand": "event_inner_location,wifi_info,location"
+          };
+          fetchObjectData(eventConstants.OBJECT_SERVICE_NAME, eventConstants.EVENT_OBJECT_NAME, queryParams, this.eventFetchSuccess.bind(this), this.eventFetchFailure.bind(this));
         }
     },
+  	
+  	 /**
+     * @function eventFetchSuccess
+     * @descripiton This function is invoked in the success response of event object
+     * @param successResponse The success response of the event object
+     * @private
+     */
+  	eventFetchSuccess : function(successResponse){
+      let eventLocationData = successResponse.hasOwnProperty("records") ? successResponse.records : null;
+      if(eventLocationData !== null){
+        let eventLocation = eventLocationData[0].location[0];
+        this.formatAndSetEventData(eventLocation);
+      }else{
+        kony.print("Fetched event data is null");
+      }
+    },
+  
+  	 /**
+     * @function eventFetchFailure
+     * @descripiton This function is invoked in the failure response of event object
+     * @param failureResponse The failure response of the event object
+     * @private
+     */
+  	eventFetchFailure : function(failureResponse){
+      kony.print(eventConstants.GENERIC_EXCEPTION_MESSAGE);
+      kony.print("Failure response is "+JSON.stringify(failureResponse));
+    },	
+  
+  	 /**
+     * @function formatAndSetEventData
+     * @descripiton This function is used to format the service data and set the data
+     * @param eventLocationData The event location data
+     * @private
+     */
+  	formatAndSetEventData : function(eventLocationData){
+       let hotelName = eventLocationData.hasOwnProperty("location") ? eventLocationData.location : "";
+       let completeAddress = this.getCompleteAddress(eventLocationData);
+       let phoneNumber = eventLocationData.hasOwnProperty("phone_number") ? eventLocationData.phone_number : "";
+       let coOrdinatesInfo = this.getCoordinates(eventLocationData);
+       this.setEventData(hotelName, completeAddress, phoneNumber, coOrdinatesInfo);
+    },
+  	
+  	 /**
+     * @function getCompleteAddress
+     * @descripiton This function is used to format and construct the complete address
+     * @param eventLocationData The event location data
+     * @private
+     */
+  	getCompleteAddress : function(eventLocationData){
+       let addressLine1 = eventLocationData.hasOwnProperty("addressLine1") ? eventLocationData.addressLine1 : "";
+       let cityName = eventLocationData.hasOwnProperty("cityname") ? eventLocationData.cityname : "";
+       let pincode = eventLocationData.hasOwnProperty("pincode") ? eventLocationData.pincode : "";
+       let countryName = eventLocationData.hasOwnProperty("country_name") ? eventLocationData.country_name : "";
+       let completeAddress = addressLine1 + ", " + cityName + ", " + pincode + ", " + countryName;
+       return completeAddress;
```

Comment:
Can't we keep concatenating to ```completeAddress``` instead of assigning to variable and then concatenating at end. Init a variable consume memory.